File: result/repair/transcoder/python2java/java_repair/MINIMUM_NUMBER_OF_JUMPS_TO_REACH_END_OF_A_GIVEN_ARRAY_2.java
  int f_gold ( int [ ] arr , int n ) {
  int [ ] jumps = new int [ n ] ;
for (int i=n-2;i>-1;i--){
    if ( ( arr [ i ] == 0 ) && ( arr [ i ] != 1 ) ) {
      jumps [ i ] = Integer.MAX_VALUE;
    }
    else if ( arr [ i ] >= n - i - 1 ) {
      jumps [ i ] = 1 ;
    }
    else {
      int min = Integer.MAX_VALUE;
      for ( int j = i + 1 ;
      j < n ;
      j ++ ) {
        if ( j <= arr [ i ] + i ) {
          if ( min > jumps [ j ] ) {
            min = jumps [ j ] ;
          }
        }
      }
            if (min!=Integer.MAX_VALUE ) {
        jumps [ i ] = min + 1 ;
      }
      else {
        jumps [ i ] = min ;
      }
    }
  }
  return jumps [ 0 ] ;
}

File: result/translate/starchat/java2cpp/c++/MINIMUM_NUMBER_OF_JUMPS_TO_REACH_END_OF_A_GIVEN_ARRAY_2.cpp
int f_gold ( int arr [ ], int n ) {
  int *jumps = new int [ n ] ;
  int min ;
  jumps [ n - 1 ] = 0 ;
  for ( int i = n - 2 ;
  i >= 0 ;
  i -- ) {
    if ( arr [ i ] == 0 ) jumps [ i ] = INT_MAX ;
    else if ( arr [ i ] >= n - i - 1 ) jumps [ i ] = 1 ;
    else {
      min = INT_MAX ;
      for ( int j = i + 1 ;
      j < n && j <= arr [ i ] + i ;
      j ++ ) {
        if ( min > jumps [ j ] ) min = jumps [ j ] ;
      }
      if ( min!= INT_MAX ) jumps [ i ] = min + 1 ;
      else jumps [ i ] = min ;
    }
  }
  return jumps [ 0 ] ;
}